### PR TITLE
refactor io 

### LINKED ISSUE #37: many things named auxincli are misleading and should be general signal client
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201632126650504/1201632305071382) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by technillogue**
fixes #37 plus some cleanups

### CODE REVIEW

**Review by itdaniher — APPROVED** [review]
rad. didn't look to see what the core conflict was, but would be happy to see this merged

File: forest/core.py [hunk 1]
```diff
@@ -192,50 +191,26 @@ async def async_shutdown(self, *_: Any, wait: bool = False) -> None:
         # call C fn _exit() without calling cleanup handlers, flushing stdio buffers, etc.
         os._exit(1)
 
-    async def handle_auxincli_raw_output(self, stream: StreamReader) -> None:
-        """Read auxin-cli output but delegate handling it"""
+    async def read_signal_stdout(self, stream: StreamReader) -> None:
+        """Read auxin-cli/signal-cli output but delegate handling it"""
```
Comment (iamalwaysuncomfortable) [inline forest/core.py:195]: Good naming

File: forest/core.py [hunk 2]
```diff
@@ -250,73 +225,82 @@ async def handle_auxincli_raw_line(self, line: str) -> None:
...
-    # In the next section, we see how the input queue is populated and consumed
+    # In the next section, we see how the outbox queue is populated and consumed
 
     pending_requests: dict[str, asyncio.Future[Message]] = {}
     pending_messages_sent: dict[str, dict] = {}
 
-    async def wait_resp(
-        self, req: Optional[dict] = None, future_key: str = ""
-    ) -> Message:
+    async def wait_resp(self, req: Optional[dict] = None, rpc_id: str = "") -> Message:
```
Comment (iamalwaysuncomfortable) [inline forest/core.py:263]: Maybe rename to "wait_for_response"?

File: forest/core.py [hunk 3]
```diff
@@ -250,73 +225,82 @@ async def handle_auxincli_raw_line(self, line: str) -> None:
...
+            await self.outbox.put(req)
         # when the result is received, the future will be set
-        response = await self.pending_requests[future_key]
-        self.pending_requests.pop(future_key)
+        response = await self.pending_requests[rpc_id]
+        self.pending_requests.pop(rpc_id)
         return response
 
-    async def auxin_req(self, method: str, **params: Any) -> Message:
+    async def signal_req(self, method: str, **params: Any) -> Message:
```
Comment (iamalwaysuncomfortable) [inline forest/core.py:281]: Maybe rename to "signal_command" (as it's the catch-all for any other command than send)

Docstring maybe something like: 
"""Sends a jsonRpc command to Signal or Auxin cli"""

**Review by iamalwaysuncomfortable — APPROVED** [review]
* Minor comments on naming

**Pull request merged**